crates/bevy_app/src/app.rs

```diff
@@ -25,11 +25,6 @@ bevy_utils::define_label!(
     AppLabelId,
 );
 
-/// The [`Resource`] that stores the [`App`]'s [`TypeRegistry`](bevy_reflect::TypeRegistry).
-#[cfg(feature = "bevy_reflect")]
-#[derive(Resource, Clone, bevy_derive::Deref, bevy_derive::DerefMut, Default)]
-pub struct AppTypeRegistry(pub bevy_reflect::TypeRegistryArc);
```

Comment:
Should we default a world to include this resource (under `bevy_reflect` feature)? Or is it still the responsibility of the user/`bevy_app` to initialize it?

I'm guessing the latter, but just want to clarify.

Reply:
That's an interesting proposition. I like the idea. Though I'm curious of the implications with regard to `Scene`s for example.

Reply:
We should avoid automatically inserting resources whenever possible :)